TikTok Ads Offline Events documentation (#1094)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/tiktok-ads-offline-events.mdx

---
title: "TikTok Ads Offline Events"
description: Send your event data from RudderStack to TikTok Ads Offline Events.
---

> [TikTok Ads](https://www.tiktok.com/business/en/apps/tiktok?tt4b_lang_redirect=1") is TikTok's online advertising platform. TikTok Ads Offline Events lets you track and send your conversions data after a user clicks your TikTok ad but does not directly proceed to sale in the online mode.

Reviewer: I understand that:
* TikTok Ads is TikTok's **online** advertising platform. 
* TikTok Ads **Offline** Events occur when a user click TikTok ad but does not directly proceed to sale in the **online** mode.

But this **online** and **offline** mentions simultaneously can be lil confusing. How about we skip first line and just explain TikTok Ads Offline Events?

Author: Makes sense.

> Find the open source code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/tiktok_ads_offline_events">GitHub repository</a>.

Reviewer: ```suggestion
Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/tiktok_ads_offline_events">GitHub repository</a>.
```

| **Connection Mode** | **Web**       | **Mobile**    | **Server**    |
| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     |  -            | -             | -             |
| **Cloud mode**      | **Supported** | **Supported** | **Supported** |

Once you have confirmed that the source platform supports sending events to TikTok Ads Offline Events, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **TikTok Ads Offline Events**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure TikTok Ads Offline Events as a destination, you will need to configure the following settings:

- **Access Token**: Enter your TikTok access token (also called Long Term Access Token) by following the steps mentioned in the [Authorization](https://ads.tiktok.com/marketing_api/docs?id=1738373141733378) and [Authentication](https://ads.tiktok.com/marketing_api/docs?id=1738373164380162) sections of the TikTok documentation.

> - **Mapping to trigger the TikTok Ads Offline standard events for the respective events**: Enter the **Event Name** and select the corresponding **Standard Event** from the dropdown to be triggered when that event is called. You can specify multiple standard events for one event name and vice versa.

Reviewer: ```suggestion
- **Mapping to trigger the TikTok Ads Offline standard events for the respective events**: Enter the **Event Name** and select the corresponding **Standard Event** from the dropdown you want to trigger when that event is called. You can specify multiple standard events for one event name and vice versa.
```

> - **Hash Contextual User Properties (SHA-256)**: Use this setting to hash the user properties like email and phone using SHA-256 encryption.

Reviewer: ```suggestion
- **Hash Contextual User Properties (SHA-256)**: Use this setting to hash the user properties like `email` and `phone` using SHA-256 encryption.
```

## Track

The <Link to="/event-spec/standard-events/track/">`track`</Link> call lets you capture the offline user events along with the associated properties.

A sample `track` call is shown below:

```javascript
rudderanalytics.track("Start Payment", {
  checkout_id: "12345",
  order_id: "1234",
  total: 20,
  revenue: 15.0,
  shipping: 22,
  tax: 1,
  discount: 1.5,
  coupon: "ImagePro",
  currency: "USD",
  contentType: "product",
  products: [{
      product_id: "123",
      sku: "G-32",
      name: "GI Joe",
      price: 4.99,
    },
    {
      product_id: "345",
      sku: "F-32",
      name: "UNO",
      price: 3.45,
      quantity: 2,
      category: "Games",
    },
  ],
})
```

### Standard events

> To send a `track` event as a standard TikTok Ads Offline Event, you must specify the event mapping in the RudderStack  dashboard setting, as shown: 

Reviewer: ```suggestion
To send a `track` event as a standard TikTok Ads Offline Event, you must specify the event mapping in the RudderStack dashboard's <Link to="#connection-settings">connection setting</Link>, as shown: 
```

<img src="../../assets/event-stream-destinations/tiktok-ads-offline-standard-events.png" alt="TikTok Ads Offline standard events" />

> For more information on this setting, refer to the <Link to="#connection-settings">Connection settings</Link> section.

Reviewer: ```suggestion
```
